frontend/components/common/CompatibleDevice.tsx

```diff
@@ -26,102 +27,128 @@ export function CompatibleDevice({
-                  ))}
-               </Flex>
-               {truncateModels && hiddenVariants.length > 0 && (
+               </chakra.a>
+            </NextLink>
+            <Flex
+               flexDir="column"
+               {...(!truncateModels && {
+                  onClick: (event) => {
+                     event.preventDefault();
+                     event.stopPropagation();
+                  },
+               })}
+               sx={{
+                  'details .whenOpen': {
+                     display: 'none',
+                  },
+                  'details[open] .whenOpen': {
+                     display: 'initial',
+                  },
+                  'details[open] .whenClosed': {
+                     display: 'none',
+                  },
+               }}
+            >
+               {visibleVariants.map((variant) => (
                   <Text
-                     mb="2px"
+                     key={variant}
                      lineHeight="short"
                      fontSize="xs"
                      color="gray.600"
                   >
-                     And {hiddenVariants.length} other model
-                     {hiddenVariants.length > 1 ? 's' : ''}...
+                     {variant}
                   </Text>
-               )}
-               {!truncateModels && hiddenVariants.length > 0 && (
-                  <Text
-                     mt="2px"
-                     lineHeight="short"
-                     fontSize="xs"
-                     fontWeight="medium"
-                     color="brand.500"
-                     {...getButtonProps()}
-                     onClick={(event) => {
-                        onToggle();
-                        event.preventDefault();
-                        event.stopPropagation();
+               ))}
+            </Flex>
+            {truncateModels && hiddenVariants.length > 0 && (
+               <Text mb="2px" lineHeight="short" fontSize="xs" color="gray.600">
+                  And {hiddenVariants.length} other model
+                  {hiddenVariants.length > 1 ? 's' : ''}...
+               </Text>
+            )}
+            {!truncateModels && hiddenVariants.length > 0 && (
+               <details>
+                  <summary
+                     style={{
+                        marginTop: '2px',
+                        lineHeight: 'short',
+                        fontSize: 'xs',
+                        fontWeight: 'medium',
+                        color: 'var(--chakra-colors-brand-500)',
+                        cursor: 'pointer',
                      }}
-                     onKeyDown={(event) => {
-                        if ('Enter' === event.code) {
-                           onToggle();
+                  >
+                     <span className="whenClosed">
+                        Show {hiddenVariants.length} more
+                     </span>
+                     <span className="whenOpen">Show fewer</span>
```

Comment:
I do like this better than toggling the strings based on the hook, though honestly I would be fine if it just said "Show X more" in both cases. I'm okay to leave it like this if you want though.

Reply:
Works for me, and makes things simpler!